tket/src/Predicates/include/Predicates/PassLibrary.hpp

```diff
@@ -123,4 +123,23 @@ const PassPtr &CnXPairwiseDecomposition();
  */
 const PassPtr &RemoveImplicitQubitPermutation();
 
+/**
+ * @brief Rewraps the internal errors of a Pass with a more suitable message.
+ * Nests the old error inside it.
+ * @param pass Compilation pass to wrap
+ * @param error_msg Message to use for the new error
+ *
+ * @tparam Error Specific error type to wrap
+ * @tparam NewError Type of the new error to throw
+ *
+ * @return compilation pass to perform this transformation
+ */
+template <typename Error, class NewError = Error>
```

Comment:
How would we wrap a pass that can throw one of several errors, each of which we want to catch?

Reply:
I was thinking of changing the signature from a string `error_msg` to a function _Error -> NewError_ (and an overloaded version for when we just want to change the string). That would let us dynamically match on multiple kinds of errors.

The problem with that is that then the wrapper cannot be serialized...